### DISCUSSION TITLE
2022-01-03 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#938] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics

- [Announcements](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900571)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900601)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900719)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900761)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900819)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900841)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900856)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900868)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1900999)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1901951)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1906096)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1907712)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/938#discussioncomment-1911360)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Announcements

- Milestone [2021-12 Improvements](https://github.com/centerofci/mathesar/milestone/32) is now closed with 13 issues completed.

**Comment by kgodey** [comment]
## Frontend priorities

1. Review and merge PRs ready for review. 
    1. #826
    2. #857 
    3. #861 
    4. #870 
    5. #931
2. Get large draft PRs reviewed and closer to merge (or merged).
    1. #915 
    2. #926
3. Continue to make progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone. There only seem to be two unblocked issues.
    1. #500 
    2. #531
4. There are [14 unassigned frontend issues](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+label%3A%22work%3A+frontend%22+milestone%3A%22[07.1]+2022-01+improvements%22+no%3Aassignee) in the [2022-01 improvements milestone](https://github.com/centerofci/mathesar/milestone/33). I think it would be good to work on some of them while we're waiting for the type-related frontend PRs to be merged and more backend work to be done on data types. Feel free to also move some into later milestones if they're not critical.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Sounds good :+1:

**Reply by pavish** [reply]
Sounds good to me.

**Comment by seancolsen** [comment]
## Sean's check in

How did your most recent work go?

- Going well. It was nice to sync up with Pavish last week and realize that (somewhat coincidentally) we've both been working on some form components that can be integrated quite nicely with each other.

What are you planning to work on soon?

- Creating front end tickets for [NULL Value Spec](https://wiki.mathesar.org/design/specs/specs-null-values) and [Foreign Key Spec](https://wiki.mathesar.org/design/specs/foreign-key-constraints).
    - I've started this and am enjoying it so far. I'm working to make the tickets as focused as possible so that some of them will be easy for new contributors to pick up.
- [Frontend work for adding multi-column unique constraints](https://github.com/centerofci/mathesar/issues/531)
- Other tickets within [2022-01 improvements](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+label%3A%22work%3A+frontend%22+milestone%3A%22%5B07.1%5D+2022-01+improvements%22+no%3Aassignee)

Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Comment by kgodey** [comment]
## Backend priorities

1. Review and merge PRs ready for review. 
    1. #865
2. Fix issues blocking the frontend:
    1. #859 (assigned to @silentninja)
    2. #932 (assigned to @dmos62)
    3. #933 (unassigned)
3. Get the following draft PRs either merged or closer to merge:
    1. Filtering
        1. #921 
        2. #879 
        3. #893
    2. Display options: #885 
4. Infrastructural issues for API error handling: 
    1. #623
    2. #883 
    3. #889
4. It would be great if someone could take these two open PRs and get them merged. They are both reviewed and waiting for updates from a contributor, but we haven't heard from them for a while. They should be fairly minor.
    1. #797 
    2. #817 
5. Continue to make progress on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)  issues.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by dmos62** [reply]
Makes sense.

**Reply by mathemancer** [reply]
I think these make sense.

**Reply by silentninja** [reply]
Makes sense

**Comment by kgodey** [comment]
## Design priorities

1. Get https://github.com/centerofci/mathesar-wiki/pull/16 closer to merge.
2. Create first draft of spec for #466 
3. Continue work on #723 

@ghislaineguerin Please comment below with thoughts/concerns (@ppii775 is out this week).

**Reply by ghislaineguerin** [reply]
@kgodey Is the plan that I start implementing certain parts of the proposed design in the prototype and @ppii775 helps with writing specs, documenting, and researching alternatives/starting discussions.

**Reply by kgodey** [reply]
@ghislaineguerin I think you should implement all of the design for #466 into the prototype and create the first draft of the spec yourself. I'm sure that there will be feedback during the initial review process and then you and @ppii775 can split up researching alternatives or ideas for different parts of the feedback. Does that make sense?

**Reply by ghislaineguerin** [reply]
@kgodey that's a good plan, yes.

**Comment by kgodey** [comment]
## Product priorities

1. Define data sources in Views better and update other Views tickets. This is starting to block design work, so this is my first priority.
    1. #806
2. Define Mathesar Types better: https://github.com/centerofci/mathesar-wiki/issues/14
3. Create issues for foreign key and `NULL` specs as needed (collaborating with @seancolsen)
4. Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-01-03 to 2022-01-07 | Ghislaine |
| 2022-01-10 to 2022-01-14 | Brent |
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |
| 2022-01-31 to 2022-02-04 | Mukesh |
| 2022-02-07 to 2022-02-11 | Sean |
| 2022-02-14 to 2022-02-18 | Dominykas |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

A bit slower than I hoped, but we have pretty good progress on frontend work for data types. The open draft PRs, once merged, should unblock all further frontend work related to data types and editing the spreadsheet.

> What are you planning to work on soon?

* Get the draft PRs for form builder and type configuration merged
* Raise PR for CellInput component for Text, Number and Boolean
* Create focused frontend tickets for each data type
* Review and merge open frontend PRs

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by dmos62** [comment]
>How did your most recent work go?

I was working on the backend filtering refactor and just before the holidays we had a call discussing major changes to that refactor.

>What are you planning to work on soon?

I'm planning to work on the above mentioned backend filtering changes. I expect the coming days to be mostly reflecting on the problem and how to solve the various conflicts.

I also picked up a minor task (https://github.com/centerofci/mathesar/issues/932), which is somewhat related to the code I was working on.

>Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No blockers.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

It went well, I just needed additional time to fix some prototype parts. Should move much faster now.

> What are you planning to work on soon?

I plan on continuing work related to Views. 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Not at the moment.

**Comment by mathemancer** [comment]
## Brent's check-in

> How did your most recent work go?

Things went pretty well before the break (as I recall).  We made progress on some tricky back end challenges.

> What are you planning to work on soon?

I want to focus mostly on PRs by others until we get the volume reduced there.  I'll look at #865, #797, and #817 to start.

I'd like to do some coding as well, so I want to jump into #933, as well as implement a number-type-only ranged grouping.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

It's possible that the ranged grouping implementation will conflict with the filtering work which heavily modifies some of the same files.  I'll catch up with @dmos62 about that.

**Reply by kgodey** [reply]
I assigned you #797, #817, and #933.

**Comment by kgodey** [comment]
> How did your most recent work go?

Before the break, I worked on wrapping up small to-do items rather than product work. It went fine, but I'd like to make some progress on product work this week.

> What are you planning to work on soon?
> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

I'll be following up on discussions, review, etc. and doing some administrative tasks, but other than that, I plan to focus on product work. No dependencies/blockers anticipated.

1. Define data sources in Views better and update other Views tickets.
2. Define Mathesar Types better.
3. Flesh out backend issues for foreign key specs.
4. Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap).

**Comment by silentninja** [comment]
## Mukesh's check-in

> How did your most recent work go?

It was slower than I expected it to be but came to a conclusion on handling the exceptions and was able to figure out where to start making the column name to id transition. 

> What are you planning to work on soon?
I want to focus on finishing up the error structure issue, followed by PR reviews.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

The column `name -> id` transition issue might interfere with filtering PR, will have a discussion with @dmos62 regrading the changes.
